core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -278,8 +278,12 @@ object DumpLogSegments {
         }
       }.mkString("{", ",", "}")
 
-      val keyString = s"metadata::$groupId"
-      val valueString = s"$protocolType:${group.protocol}:${group.generationId}:$assignment"
+      val keyString = Json.encode(Map("metadata" -> groupId))
+      val valueString = Json.encode(Map(
+          "protocolType" -> protocolType,
+          "groupMetadata.protocol" -> group.protocol,
```

Comment:
Out of curiosity, why do we use the "groupMetadata" prefix?

Reply:
Hmm, it's been a while :) Perhaps to somehow reflect the object/value they are describing. I could remove the prefix in the next update.